src/fondant/dataset/dataset.py

```diff
@@ -472,6 +484,9 @@ def register_operation(
         input_dataset: t.Optional["Dataset"],
         output_dataset: t.Optional["Dataset"],
     ) -> None:
+        if self._graph is None:
+            self._graph = OrderedDict()
```

Comment:
Is this necessary? `self._graph` is created as an `OrderedDict` in the `__init__` method.